src/Symfony/Component/Finder/Iterator/FilterIterator.php

```diff
@@ -18,9 +18,24 @@
  *
  * @author Alex Bogomazov
  */
-abstract class FilterIterator extends \FilterIterator
+abstract class FilterIterator extends \FilterIterator implements \RecursiveIterator
 {
     /**
+     * @var bool The memoized return value of $this->accept() when having a recursive inner iterator makes sense.
```

Comment:
is it really needed ? RecursiveIteratorIterator will call hasChildren only for the current entry, which means that ``accept`` will have accepted them already